src/allmydata/test/test_openmetrics.py

```diff
@@ -0,0 +1,118 @@
+           'storage_server.latencies.writev.50_0_percentile': 0.0004138946533203125,
+           'storage_server.latencies.read.95_0_percentile': 9.107589721679688e-05,
+           'storage_server.latencies.readv.95_0_percentile': 0.0003859996795654297,
+           'storage_server.latencies.write.10_0_percentile': 3.719329833984375e-05,
+           'storage_server.accepting_immutable_shares': 1,
+           'storage_server.latencies.writev.samplesize': 309,
+           'storage_server.latencies.get.95_0_percentile': 0.0003190040588378906,
+           'storage_server.latencies.readv.10_0_percentile': 0.00032210350036621094,
+           'storage_server.latencies.get.90_0_percentile': 0.0002999305725097656,
+           'storage_server.latencies.get.01_0_percentile': 0.0001239776611328125,
+           'cpu_monitor.total': 641.4941180000001,
+           'storage_server.latencies.write.samplesize': 1000,
+           'storage_server.latencies.write.95_0_percentile': 9.489059448242188e-05,
+           'storage_server.latencies.read.50_0_percentile': 6.890296936035156e-05,
+           'storage_server.latencies.writev.01_0_percentile': 0.00033211708068847656,
+           'storage_server.latencies.read.10_0_percentile': 3.0994415283203125e-05,
+           'storage_server.latencies.allocate.10_0_percentile': 0.0004949569702148438,
+           'storage_server.reserved_space': 0,
+           'storage_server.disk_avail': 103289454592,
+           'storage_server.latencies.write.99_0_percentile': 0.00011301040649414062,
+           'storage_server.latencies.write.90_0_percentile': 9.083747863769531e-05,
+           'cpu_monitor.5min_avg': 0.0002370666691157502,
+           'storage_server.latencies.write.mean': 5.8008909225463864e-05,
+           'storage_server.latencies.readv.50_0_percentile': 0.00033020973205566406,
+           'storage_server.latencies.close.99_0_percentile': 0.0004038810729980469,
+           'storage_server.allocated': 0,
+           'storage_server.latencies.writev.99_0_percentile': 0.0007710456848144531,
+           'storage_server.latencies.readv.99_9_percentile': 0.0004780292510986328,
+           'storage_server.latencies.read.samplesize': 170,
+           'storage_server.latencies.allocate.samplesize': 406,
+           'storage_server.latencies.allocate.95_0_percentile': 0.0008411407470703125
+         }, 'counters': {
+           'storage_server.writev': 309,
+           'storage_server.bytes_added': 197836146,
+           'storage_server.close': 326,
+           'storage_server.readv': 14299,
+           'storage_server.allocate': 406,
+           'storage_server.read': 170,
+           'storage_server.write': 3775,
+           'storage_server.get': 472}
+         }
+        return stats
+
+class OpenMetrics(unittest.TestCase):
+    def test_spec_compliance(self):
+        """
+        Does our output adhere to the OpenMetrics spec?
+        https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md
```

Comment:
I wonder if there is any more public/stable version of this URL.  I imagine that in a month or two someone will decide to re-organize the whole OpenMetrics repo and this one will break.

Reply:
Hope you like the new links better!